src/components/WeatherWidget/WeatherWidget.jsx

```diff
@@ -0,0 +1,137 @@
+import { StyledWrapper, StyledWrapperWeather } from "./WeatherWidget.styles";
+import { useState, useEffect } from "react";
+import getLocation from "../../services/location";
+import getWeather from "../../api/weather";
+import CircularProgress from "@mui/material/CircularProgress";
+import Box from "@mui/material/Box";
+import { Switch, FormControlLabel } from "@mui/material";
+
+const weatherImages = {
+  rainy: {
+    condition: (data) => data.precipitation_probability > 50,
+    image: "rainy-day",
+    description: "Rainy image",
+  },
+  snowy: {
+    condition: (data) => data.snowfall > 0,
+    image: "snow",
+    description: "Snowy image",
+  },
+  windy: {
+    condition: (data) => data.windspeed > 20,
+    image: "wind",
+    description: "Windy image",
+  },
+  sunny: {
+    condition: (data) => data.temperature > 20,
+    image: "sun",
+    description: "Sunny image",
+  },
+  cloud: {
+    condition: () => true,
+    image: "cloud",
+    description: "Cloudy image",
+  },
+};
+
+export const WeatherWidget = () => {
+  const [weatherData, setWeatherData] = useState(null);
+  const [error, setError] = useState(null);
+  const [temperatureUnit, setTemperatureUnit] = useState("C");
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchWeather = async () => {
+      try {
+        const location = await getLocation();
+        const weather = await getWeather(location);
+        setWeatherData(weather);
+      } catch (err) {
+        setError(err.message);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchWeather();
+  }, []);
+
+  const getWeatherImage = () => {
+    if (!weatherData) {
+      return {
+        image: `${process.env.PUBLIC_URL}/images/cloud.png`,
+        description: "Cloudy weather",
+      };
+    }
+    for (const key in weatherImages) {
+      if (weatherImages[key].condition(weatherData)) {
+        return {
+          image: `${process.env.PUBLIC_URL}/images/${weatherImages[key].image}.png`,
+          description: weatherImages[key].description,
+        };
+      }
+    }
```

Comment:
`for` is good here.

Just sharing for you that

```
Object.keys(weatherImages).map((key) => ...)
```

would be an option too. Maybe tiny little bit nicer. Let's talk tomorrow about processing arrays.